jira.py

```diff
@@ -1,37 +1,74 @@
 from errbot import BotPlugin
-import logging
 import re
 import requests
+import sys
 from requests_kerberos import HTTPKerberosAuth, DISABLED
+from itertools import chain
 
+CONFIG_TEMPLATE = {'URL': "http://jira.example.com",
+                   'PROJECTS': ['FOO', 'BAR'],
+                   'KERBEROS': True,
+                   'USERNAME': 'foo',
+                   'PASSWORD': 'bar',
+                   'SSL_VERIFY': True
+                  }
 
 class Jira(BotPlugin):
     """A plugin for interacting with Atlassian JIRA"""
-    min_err_version = '1.6.0'  # Optional, but recommended
-    max_err_version = '2.0.0'  # Optional, but recommended
 
     def get_configuration_template(self):
         """Defines the configuration structure this plugin supports"""
-        return {'URL': "http://jira.example.com",
-                'PROJECTS': ['FOO', 'BAR']}
+        return CONFIG_TEMPLATE
+
+    def configure(self, configuration):
+        """
+        Creates a Python dictionary object which contains all the values from our
+        CONFIG_TEMPLATE and then updates that dictionary with the configuration
+        received when calling the "!plugin config JIRA" command.
+        """
+
+        if configuration is not None and configuration != {}:
+            config = dict(chain(CONFIG_TEMPLATE.items(),
+                                configuration.items()))
+        else:
+            config = CONFIG_TEMPLATE
+        super(Jira, self).configure(config)
+
 
     def get_cookie(self):
-        r = requests.get(self.config['URL']+'/step-auth-gss',
-                         auth=HTTPKerberosAuth(mutual_authentication=DISABLED),
-                         verify=False)
-        return r.cookies['JSESSIONID']
+        try:
+            cookie = self['COOKIE']
+        except KeyError:
+            self.log.debug("[JIRA] No JIRA cookie found, requesting one...")
```

Comment:
Not sure about this one, cookie might get expired. We probably should be moving this part to a separate function and use `start_poller` to refresh the cookie.

Anyway, I'm gonna test it and see if it actually happens.